src/modules/product/services/product.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from '@nestjs/common';
+import { Brackets } from 'typeorm';
+
+import { ProductEntity } from 'src/entities/product.entity';
+import { ProductsFilterDto } from 'src/modules/product/dto/filter-products.dto';
+import { ProductRepository } from 'src/modules/repository/services/product.repository';
+
+@Injectable()
+export class ProductService {
+  constructor(private readonly productRepository: ProductRepository) {}
+
+  public async getAllProducts(
+    dto: ProductsFilterDto,
+  ): Promise<{ totalCount: number; products: ProductEntity[] }> {
+    const {
+      page = 1,
+      limit = 5,
+      search = '',
+      size,
+      style,
+      color,
+      priceRange,
+      order = 'ASC',
+    } = dto;
+
+    const queryBuilder = this.productRepository.createQueryBuilder('product');
+
+    if (search) {
+      queryBuilder.andWhere(
+        new Brackets((qb) => {
```

Comment:
maybe we should move this to a repository

Reply:
done